adapter/src/main/java/ar/edu/undec/adapter/service/curso/config/CursoBeanConfig.java

```diff
@@ -11,12 +13,12 @@
 public class CursoBeanConfig {
 
     @Bean
-    public CrearCursoUseCase crearCursoUseCase(CrearCursoRepository crearCursoRepository, BuscarCursoPorIdRepository buscarCursoPorIdRepository) {
-        return new CrearCursoUseCase(crearCursoRepository, buscarCursoPorIdRepository);
+    public CrearCursoInput crearCursoInput(CrearCursoRepoImplementation crearCursoRepoImplementation, BuscarCursoPorIdRepoImplementation buscarCursoPorIdRepoImplementation) {
```

Comment:
Aquí no había que cambiar los parámetros Repository 😅, esos estaban bien siendo Interfaces. En la firma del método, procurar recibir y devolver abstracciones. Al momento de ejecutar, éstas serán reemplazadas por las respectivas implementaciones, pero en la firma procuremos definir interfaces
```suggestion
    public CrearCursoInput crearCursoInput(CrearCursoRepository crearCursoRepository, BuscarCursoPorIdRepository buscarCursoPorIdRepository) {
```